docs/dev_guide/design_doc.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: default
+title: Design Documents
+parent: Developer Guides
+permalink: /docs/dev_guide/design_doc
+---
+# Design Documents
+
+## Venice Improvement Proposal (VIP)
+
+Venice follow the Venice Improvement Proposal (VIP). VIP is the mechanism used to propose changes to the 
+Venice Codebase. 
+
+Not all the changes require a VIP treatment, Please use your commonsense to check if writing this VIP help yourself 
+and the reviewers time instead of just doing it in the pull request itself. Generally, this applies to
+* Large code refactors
+* New features
+* Public API changes
+
+In practical terms, the PIP defines a process in which developers can submit a design doc, 
+receive feedback and get the "go ahead" to execute.
+
+## Who can create a VIP?
+
+Any person willing to contribute to the Venice project is welcome to
+create a VIP.
+
+## How does the VIP process work?
+
+A VIP proposal can be in these states:
+1. **DRAFT**: (Optional) This might be used for contributors to collaborate and
+   to seek feedback on an incomplete version of the proposal.
+
+2. **DISCUSSION**: The proposal has been submitted to the community for
+   feedback and approval.
+
+3. **ACCEPTED**: The proposal has been accepted by the Venice project.
+
+4. **REJECTED**: The proposal has not been accepted by the Venice project.
+
+5. **IMPLEMENTED**: The implementation of the proposed changes have been
+   completed and everything has been merged.
+
+5. **RELEASED**: The proposed changes have been included in an official
+   Venice release.
+
+
+The process works in the following way:
+
+1. The author(s) of the proposal will create a file named "VIP-xxx.md" in [proposal](../../docs/proposals) folder cloning 
+    from the [template for VIP proposals](../../docs/proposals/VIP_TEMPLATE.md). The "xxx" number should be chosen to 
+    be the next number from the existing PIP issues, listed [here](../../docs/proposals/)
+2. The author(s) submit this file as a PR named "VIP-xxx: {short description}".
+3. People discuss using PR comments, each is its own threaded comment. 
+   General comments can be made as general comment in the PR.
+4. Comment was done? They resolve it. This way you see what the pending discussions are at a glance.
+5. PIP comments resolved or no more concerns? Awesome. A Venice maintainer will merge the PR. 
+    Merge == Approved 
+6. Rejected ? Close the PR 
```

Comment:
I think that doesn't work so well, because then the VIP-XXX.md file won't get committed to the repo. I think all of the PRs should be merged (unless perhaps they are discarded out of hand due to not respecting the template or other basic issues...). The status of the PR does not need to align to the status of the VIP.

Reply:
So the proposal in the Pulsar  was to close the rejected PR, this is not same as deleting it. It will be around as a closed PR. Looking for all the rejected proposal -- search pr:is_closed label:vip. 

But I think we can do it differently. In the design_doc.md page (landing page) we can categorize them into ACCEPTED and REJECTED categories. The issue with this approach is that we have to maintain the bookkeeping. Closing the PR avoids that bookkeeping efforts. 

This is why I went with the merged/closed approach so that its light weight on the process side.

Reply:
I guess I'm envisioning the following workflow:

1. The proposer submits a PR with the VIP in either DRAFT or DISCUSSION stage.
2. We do a first round of discussion on the PR. Depending on the outcome of the discussion, the status could move to ACCEPTED or REJECTED, or it could stay in DISCUSSION stage (e.g. if we agree tentatively on the broad strokes, but there are still action items to refine certain aspects). At the end of this, the PR gets merged, and at that point the VIP will appear in the directory of all historical VIPs.
3. If the merged VIP is still in DISCUSSION stage, then further PRs will be submitted to address the remaining action items.
4. If the VIP is ACCEPTED, then next stages are implementation, with eventually some code change PR also carrying an alteration of the VIP page to move the status to IMPLEMENTED.

End result is that you can look at the directory of VIPs and see all of them, both accepted and rejected, and even some which are still in discussion stage. Only the ones which are still in the first round of PR review would not show up in the directory.

LMK if that makes sense.

Reply:
Sounds good. Updated the instructions as well.